### ISSUE TITLE
`required` and `not` field missing for non-properties object

### ISSUE THREAD

**Issue by xunleii** [dadav/helm-schema#39]
Hi, firstly, thanks for this tool; it really simplifies the generation on JsonSchema.

However, I have a problem with the schema definition; I'd like to generate something like
```json
...
"not" : {
  "anyOf" : [
    { "required" : ["aaa"] },
    { "required" : ["bbb"] }
  ]
}
...
```
to filter a field coming from a `$ref` but I can't at the moment because `not` is missing and `required` can only be generated inside a `properties` object _(if I understood https://github.com/dadav/helm-schema/blob/main/pkg/schema/schema.go#L359 correctly)_.

Is it possible to add them to this tool ?

_NOTE: I tried [this](https://github.com/chezmoi-sh/helm-schema/commit/93ee7a52b5ac95ff96b632dc31a84960f66b998c) which work as expected; if you want, I can create a PR to add them with few test cases to validate the behavior_

**Comment by xunleii** [comment]
Thanks for adding the `not` keyword!

**Comment by dadav** [comment]
It's ok, we can still use this issue.

**Comment by dadav** [comment]
Should be fixed in #55
